### DISCUSSION TITLE
Where do I find the source code for the Enable-TlsCipherSuite cmdlet?

### DISCUSSION THREAD

**Discussion by ErikHumphrey** [PowerShell/PowerShell#18936] [Q&A] [answered]
I'm looking for the source code for the [`Enable-TlsCipherSuite`](https://learn.microsoft.com/en-us/powershell/module/tls/enable-tlsciphersuite?view=windowsserver2022-ps) cmdlet. I'm guessing I can find it somewhere in Windows, but I don't have a Windows machine handy to check.

1. Is `Enable-TlsCipherSuite` open-source? Where can I find its source code on GitHub?
2. Is the code for this cmdlet viewable somewhere in Windows? Where? I haven't tried `Get-Command` yet.

I noticed the source code for some cmdlets seems to be in the repository, but maybe not all of them, or they're under different names.

**Comment by iSazonov** [comment, marked as answer]
The cmdlet is in a module which is a part of Windows OS and it is not open source.

The repository contains source code for pwsh (PowerShell Core) and for some built-in core modules. Most of other modules is not in the repository and have other owners (in MSFT or third party).

**Comment by jborean93** [comment]
Adding to the above, it's a binary cmdlet and calls some Win32 functions through PInvoke. I've reimplemented the `Get-TlsCipherSuite` in a PowerShell function that you can see here https://gist.github.com/jborean93/adec1d69e35b05be23abed32eb6063f4. The `Enable` action I believe is calling [BCryptAddContextFunction](https://learn.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptaddcontextfunction). I believe you can even manually edit the registry key which stores this confirmation for you.
